shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/controllers/BridgeExecutorController.java

```diff
@@ -158,10 +162,14 @@ public DeleteControl cleanup(BridgeExecutor bridgeExecutor, Context context) {
         return DeleteControl.defaultDelete();
     }
 
-    private void notifyDeploymentFailure(BridgeExecutor bridgeExecutor, String failureReason) {
-        LOGGER.warn("Processor deployment BridgeExecutor: '{}' in namespace '{}' has failed with reason: '{}'",
-                bridgeExecutor.getMetadata().getName(), bridgeExecutor.getMetadata().getNamespace(), failureReason);
-        notifyManager(bridgeExecutor, ManagedResourceStatus.FAILED);
+    @Override
+    public Optional<BridgeExecutor> updateErrorStatus(BridgeExecutor bridgeExecutor, RetryInfo retryInfo, RuntimeException e) {
```

Comment:
This is my implementation of `ErrorStatusHandler`.

If k8s failed to provision something I update the CRD status and notify the `manager.